### ISSUE TITLE
`std::arch::wasm::unreachable()` is stable but does not compile

### ISSUE THREAD

**Issue by eric-seppanen** [rust-lang/rust#133015]
I tried to build this code with `cargo build --target=wasm32-unknown-unknown`:

```rust
pub fn force_wasm_trap() {
    core::arch::wasm::unreachable();
}
```

I expected to see this happen: compile succeeds, generates a function containing the wasm "unreachable" instruction, which is what [the documentation says it does](https://doc.rust-lang.org/core/arch/wasm/fn.unreachable.html).

Instead, this happened:
```plaintext
error[E0658]: use of unstable library feature 'simd_wasm64'
 --> src/lib.rs:2:5
  |
2 |     std::arch::wasm::unreachable();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #90599 <https://github.com/rust-lang/rust/issues/90599> for more information

For more information about this error, try `rustc --explain E0658`.
```

I don't understand what happened: the documentation says this has been stable since rust 1.37.0, and I don't think there's any reason why the webassembly "unreachable" instruction would be tied to the "simd_wasm64" feature.

I also tried with and without `#[no_std]` and using both `core::arch` and `std::arch`: all fail the same way.

### Meta

`rustc --version --verbose`:
```
rustc 1.82.0 (f6e511eec 2024-10-15)
binary: rustc
commit-hash: f6e511eec7342f59a25f7c0534f1dbea00d01b14
commit-date: 2024-10-15
host: x86_64-unknown-linux-gnu
release: 1.82.0
LLVM version: 19.1.1
```

The same error happens with `1.83.0-beta.5` and `nightly-2024-11-12`.

**Comment by eric-seppanen** [comment]
[playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3cb90cf13786f77002da6e88ad4a5960), which also shows the error when set to compile wasm.

**Comment by lukas-code** [comment]
This is intentional: The `wasm64` intrinsics are still unstable and therefore the [`std::arch::wasm`](https://doc.rust-lang.org/nightly/core/arch/wasm/index.html) and [`std::arch::wasm64`](https://doc.rust-lang.org/nightly/core/arch/wasm64/index.html) modules are also unstable. This means that the function `std::arch::wasm::unreachable` isn't be accessible on stable, because a component of the path is still unstable.

The function `std::arch::wasm::unreachable` only appears as stable in the docs, because it's a re-export of [`std::arch::wasm32::unreachable`](https://doc.rust-lang.org/nightly/core/arch/wasm32/fn.unreachable.html), which is actually properly stable in 1.37.0 -- The incorrect stability display is a bug that has been fixed and the correct stability will show if you look at the [nightly docs of `std::arch::wasm::unreachable`](https://doc.rust-lang.org/nightly/core/arch/wasm/fn.unreachable.html).

**Comment by SNCPlay42** [comment]
The diagnostic could still use improvement to actually indicate which part of the path is the problem.

**Comment by eric-seppanen** [comment]
Right, got a bit confused about the difference between `arch::wasm` and `arch::wasm32`. I will use `core::arch::wasm32::unreachable` instead.

**Comment by eric-seppanen** [comment]
I can close this, unless someone wants to hold onto it for docs/diagnostics reasons?

**Comment by lukas-code** [comment]
Keeping this open for improving the diagnostic seems reasonable, it would probably help to shorten the span of the path to the segment that's actually unstable:

```
error[E0658]: use of unstable library feature 'simd_wasm64'
 --> src/lib.rs:2:5
  |
2 |     std::arch::wasm::unreachable();
  |                ^^^^
  |
  = note: see issue #90599 <https://github.com/rust-lang/rust/issues/90599> for more information

For more information about this error, try `rustc --explain E0658`.
```

**Comment by CryZe** [comment]
Instead of improving the diagnostic, this could also just be stabilized, as the proposal reached phase 4 and is about to ship in Chrome and Firefox.

**Comment by CodesInChaos** [comment]
@CryZe the diagnostic improvement would affect all unstable modules, so it'd still be useful if this gets stabilized.

**Comment by alexcrichton** [comment]
I believe that this is an expected error at the moment, and the appropriate tracking issue for this is over at https://github.com/rust-lang/rust/issues/90599.
